Framework/Algorithms/src/XDataConverter.cpp

```diff
@@ -69,17 +69,19 @@ void XDataConverter::exec() {
   RebinnedOutput_sptr outRB = std::dynamic_pointer_cast<RebinnedOutput>(outputWS);
   // Force fractions to unity (converting from histo to point discards bin info).
   MantidVecPtr outF;
+  // need this line to prevent a hard crash
```

Comment:
I think the declaration of outF needs to be moved into the loop at~L79 as well. At the moment you may have different threads in the parallel if statement lower down resizing the single outF vector to different sizes. If the spectra don't have the same number of bins this will cause a problem. Moving the declaration inside the loop will mean that each thread has its own copy of outF.
If you do this I think you'll be able to remove the extra resize call on L73

Reply:
I think Anthony has just done that in his last commit. The code looks fine to me now. Will approve shortly once it's passed a couple more of the jobs

Reply:
Ah, I meant something like:

```
    MantidVecPtr outF;

    // Copy over the Y and E data
    outputWS->setSharedY(i, inputWS->sharedY(i));
    outputWS->setSharedE(i, inputWS->sharedE(i));
    if (isRebinnedWorkspace) {
      outF.access().resize(inputWS->getNumberBins(i), 1.0);
      if (outRB) {
         outRB->setF(i, outF);
      }
    }
```

sorry for messing ip the spacings (writing this on my phone)...

Reply:
If the first part of an if statement fails for an "and", it does not evaluate the second part (as it fails as a whole anyway)

Reply:
Yes, sorry I have misread outRB as outF so rearranged the ifs but its not necessary as you say...